test/unit/user_test.rb

```diff
@@ -178,5 +178,18 @@ class UserTest < ActiveSupport::TestCase
     assert_not basic_user.moderator?
     assert_not basic_user.can_moderate?
   end
-
+  test 'user email validation' do
+    user = User.new(username: 'zen',
+                    password: 'nez',
+                    password_confirmation: 'nez',
+                    email: 'abc@xyz.com')
```

Comment:
This email is valid and the user should be saved to the database. 
What do you think @SidharthBansal?

Reply:
Sorry I missed, you need to write it should not save to the database.
@seafr thanks